internal/workload/manager.go

```diff
@@ -175,13 +191,13 @@ func (w *WorkloadManager) Update(configuration models.DeviceConfigurationMessage
 		if _, ok := configuredWorkloadNameSet[name]; !ok {
 			log.Infof("Workload not found: %s. Removing", name)
 			manifestPath := w.getManifestPath(name)
-			err := os.Remove(manifestPath)
-			if err != nil {
-				if !os.IsNotExist(err) {
-					errors = multierror.Append(errors, fmt.Errorf("cannot remove existing manifest workload: %s", err))
-				}
+			if err := deleteFile(manifestPath); err != nil {
+				errors = multierror.Append(errors, fmt.Errorf("cannot remove existing manifest workload: %s", err))
+			}
+			authPath := w.getAuthFilePath(name)
+			if err := deleteFile(authPath); err != nil {
+				errors = multierror.Append(errors, fmt.Errorf("cannot remove existing workload auth file: %s", err))
```

Comment:
what happens if the manifest gets deleted and the secret not ? we won't be able to track this file that needs to be deleted IIUC.

Would a folder structure per workload will be easier to manage in that case ?

Reply:
Folder structure is a great idea for that.
I will make sure that the file is in expected state in the `ensureWorkloadsFromManifestsAreRunning` function and we can refactor the file structure in a separate PR - there would be many changes unrelated to this one.